### PR TITLE
Response to Diagnostic.

### PR DESCRIPTION
**Pull request by dumplingpuff**

### CODE REVIEW

**Comment by dumplingpuff** [comment]
Comfort: 4,
Clarity: 3.5
I used ember docs and class notes for my answer.  Mostly Ember docs.

File: diagnostic.md [hunk 1]
```diff
@@ -6,21 +6,28 @@ Record your responses inside the fenced code blocks below each question.
     and what are the main task(s) you perform inside an Ember Route?
 
     ```md
-    <!-- your response here -->
+    The main task you do in router is to set routes or url paths to the correct
+    route file.  In the route file that is where you define your model or data.
```
Comment (faetea) [inline diagnostic.md:10]: Yes, the Ember Application Router defines the route, and the Ember Route specifies how to load model data using the model hook.

File: diagnostic.md [hunk 2]
```diff
@@ -6,21 +6,28 @@ Record your responses inside the fenced code blocks below each question.
...
+
     ```
 
 1.  What is the command to generate a route named `boston` nested under
     `campus`?
 
     ```md
-    <!-- your response here -->
+    ember g route campus/boston
```
Comment (faetea) [inline diagnostic.md:18]: yes

File: diagnostic.md [hunk 3]
```diff
@@ -6,21 +6,28 @@ Record your responses inside the fenced code blocks below each question.
...
     use the `link-to` helper to generate an appropriate link?
 
     ```md
-    <!-- your response here -->
+    What template are you linking from is it application or the campus template?
+    campus:
+    {{#link-to 'boston'}}{{/link-to}}
+
+    application:
+    {{#link-to 'campus.boston'}}{{/link-to}}
```
Comment (faetea) [inline diagnostic.md:30]: We want to use the second link-to: `'campus.boston'`.

File: diagnostic.md [hunk 4]
```diff
@@ -35,7 +42,13 @@ Record your responses inside the fenced code blocks below each question.
...
 
     ```md
-    <!-- your response here -->
+    The first route is saying that it is a nested route and that product model will be listed under the products model.
+    What happens with the first route is that the products model
+    will first render and then the product model will render inside
+    of it where there is an outlet.
+
+    The latter is not nested but still a dynamic route, product model
+    will load independently of products model.
```
Comment (faetea) [inline diagnostic.md:51]: Looks Good!  These two route definitions are identical URL paths.  Nested route definitions, do require a nested file structure, and the big distinction is where Ember will render the content.  
The nested `'products'/'product'` instructs Ember to render
`app/products/product/template.hbs` in the `outlet` of the parent `app/products/template.hbs`.  
The non-nested `'product'` instructs Ember to render `app/product/template.hbs` in the context of the application template.

File: diagnostic.md [hunk 5]
```diff
@@ -48,11 +61,13 @@ Record your responses inside the fenced code blocks below each question.
     value `'123'` inside a Route?
 
     ```md
-    <!-- your response here -->
+    this.store.find('movies', params.movie_id);
```
Comment (faetea) [inline diagnostic.md:64]: yes

File: diagnostic.md [hunk 6]
```diff
@@ -48,11 +61,13 @@ Record your responses inside the fenced code blocks below each question.
...
     ```
 
 1.  Inside a template, how do we reference data provided by a Route?
 
     ```md
-    <!-- your response here -->
+      We want to hook the model data into the template.
+      {{#each model as |movie|}}
+      {{/each}}
```
Comment (faetea) [inline diagnostic.md:72]: yes

**Comment by faetea** [comment]
6/6

Great Job, you're studying really hard! :+1:

**Pull request closed without merging**